python/cudf/cudf/core/index.py

```diff
@@ -1543,6 +1558,23 @@ def _indices_of(self, value):
         """Return indices of value in index"""
         return self._column.indices_of(value)
 
+    def __dask_tokenize__(self):
+        try:
+            # Import dask_cudf (if available) in case
+            # this is being called within Dask Dataframe
+            import dask_cudf  # noqa: F401
+
+        except ImportError:
+            warnings.warn(
+                f"Using dask to tokenize a {type(self)} object, "
+                f"but `dask_cudf` is not installed. Please install "
+                f"`dask_cudf` for proper dispatching."
+            )
+
+        # TODO: Avoid `to_pandas` once gpu hashing can
+        # produce a single (deterministic) token
+        return (type(self), self.to_pandas())
+
```

Comment:
```suggestion
```

Rationale: `GernericIndex` inherits from `SingleColumnFrame` which already implements this.

Reply:
That said, an index is immutable so that probably means there's a better way of doing this.